src/internal/operators/take.ts

```diff
@@ -45,26 +46,50 @@ import { OperatorSubscriber } from './OperatorSubscriber';
  * the source if the source emits fewer than `count` values.
  */
 export function take<T>(count: number): MonoTypeOperatorFunction<T> {
-  return count <= 0
-    ? // If we are taking no values, that's empty.
-      () => EMPTY
-    : operate((source, subscriber) => {
-        let seen = 0;
-        source.subscribe(
-          new OperatorSubscriber(subscriber, (value) => {
-            // Increment the number of values we have seen,
-            // then check it against the allowed count to see
-            // if we are still letting values through.
-            if (++seen <= count) {
-              subscriber.next(value);
-              // If we have met or passed our allowed count,
-              // we need to complete. We have to do <= here,
-              // because re-entrant code will increment `seen` twice.
-              if (count <= seen) {
-                subscriber.complete();
-              }
-            }
-          })
-        );
-      });
+  if (count < 1) {
+    // If we are taking no values, that's empty.
+    return () => EMPTY;
+  }
+
+  if (count === Number.POSITIVE_INFINITY) {
+    return identity;
+  }
+
+  // this condition also returns true on left-side NaN and not just if count
+  // exceeds MAX_SAFE_INTEGER
+  if (!(count < Number.MAX_SAFE_INTEGER + 1)) {
```

Comment:
actually, this was done intentionally, since `Number.MAX_SAFE_INTEGER + somethingSmallerThanOne` is ok, (gets floored after). but it could very well be that there is no number representable between MAX_SAFE_INTEGER and MAX_SAFE_INTEGER + 1 in 64bit floats

Reply:
```suggestion
  if (Number.MAX_SAFE_INTEGER < count) {
```

1. Inverting the check makes it less readable.
2. Accounting for `Number.MAX_SAFE_INTEGER + 0.56` or whatever won't make a difference, because you can't take 0.56 of something :)